src/interp/binary-reader-interp.cc

```diff
@@ -856,6 +851,16 @@ Result BinaryReaderInterp::EndFunctionBody(Index index) {
 Result BinaryReaderInterp::OnLocalDeclCount(Index count) {
   local_decl_count_ = count;
   local_count_ = 0;
+  // FIXME(Soni): does the value of `values` even matter here? it used to be
+  // always 0 when this call was in BeginFunctionBody.
+  // NOTE: we don't count the parameters, as they're not part of the frame.
+  func_->handlers.push_back(HandlerDesc{HandlerKind::Catch,
+                                        istream_.end(),
+                                        Istream::kInvalidOffset,
+                                        {},
+                                        {Istream::kInvalidOffset},
+                                        static_cast<u32>(local_decl_count_),
+                                        0});
```

Comment:
Why did this code move?

Reply:
fix for #2476 (you may want us to split this out, along with the fix for #2469 ?). we implemented the fix while trying to figure out how to work the interpreter (unfortunately we still don't know how to implement EHv4 in the interpreter)

Reply:
Yup, lets split out whatever we can.